js/stats.js

```diff
@@ -0,0 +1,107 @@
+require.scopes.stats = ( () => {
+    var utils = require('utils'),
+        statsByParentCompany = {},
+        statsBySite = {},
+        topBlocked = [];
```

Comment:
would be better to use a data structure designed for this purpose like a heap, designed to return the largest n values. eg http://www.collectionsjs.com/heap

https://en.wikipedia.org/wiki/Selection_algorithm